src/ast/FieldDecl.ts

```diff
@@ -12,12 +12,41 @@ export class FieldDecl extends AstNode {
 
     fields: VarList;
     modifier: string;
-    generateGetter: boolean;
-    generateSetter: boolean;
+    generateGetter: boolean = false;
+    generateSetter: boolean = false;
 
     public parse(context: Tokenizer): any {
-        // TODO: implement the rest.
+        let indentLevel: number = context.getCurrentLineTabLevel();
+        context.getAndCheckNext("fields");
+        this.modifier = context.getNext();
         this.fields = new VarList();
+        this.fields.parse(context);
+
+        context.getNext();
```

Comment:
we probably want to grab this inside the VarList node, as this would get the last "]" token rather than here (after the termination of the while loop, the last thing we do in VarList.parse()).